be/src/formats/parquet/column_reader.cpp

```diff
@@ -588,52 +646,21 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ParquetField*
         RETURN_IF_ERROR(reader->init(field, std::move(key_reader), std::move(value_reader)));
         *output = std::move(reader);
     } else if (field->type.type == LogicalType::TYPE_STRUCT) {
-        // For Struct type, we need consider subfield not existed suitition.
-        // When Iceberg add a new struct subfield, the original parquet file do not contains newly added subfield,
-        // so we will append default value for it.
-        std::unordered_map<std::string, const TIcebergSchemaField*> subfield_name_2_field_schema{};
-        for (const auto& each : iceberg_schema_field->children) {
-            std::string format_subfield_name =
-                    opts.case_sensitive ? each.name : boost::algorithm::to_lower_copy(each.name);
-            subfield_name_2_field_schema.emplace(format_subfield_name, &each);
-        }
-
-        std::unordered_map<int32_t, size_t> field_id_2_pos{};
-        for (size_t i = 0; i < field->children.size(); i++) {
-            field_id_2_pos.emplace(field->children[i].field_id, i);
-        }
+        std::vector<int32> subfield_pos(col_type.children.size());
+        std::vector<const TIcebergSchemaField*> iceberg_schema_subfield(col_type.children.size());
```

Comment:
`new_iceberg_schema_field` instead of `iceberg_schema_subfield`?

Reply:
And I think maybe we don't need to copy a `std::vector<const TIcebergSchemaField*>`, just use the original one.